### DISCUSSION TITLE
Question: Does sarama support self-signed certificates with verification enabled?

### DISCUSSION THREAD

**Discussion by wcallgair** [IBM/sarama#2623] [Q&A] [answered]
I have a kafka cluster with self-signed certificates and this works fine with java consumers, however my apps that use sarama cannot connect with tls verification enabled. Ive seen posts from other communities saying they had to disable tlsverify. That dosnt seem correct.. Does sarama support mutual tls with self-signed certificates?

Thanks

**Comment by ghost** [comment]
Thank you for taking the time to raise this issue. However, it has not had any activity on it in the past 90 days and will be closed in 30 days if no updates occur.
Please check if the master branch has already resolved the issue since it was raised. If you believe the issue is still valid and you would like input from the maintainers then please comment to ask for it to be reviewed.

**Comment by dnwe** [comment, marked as answer]
@wcallgair yes Sarama just accepts a standard Go [tls.Config](https://pkg.go.dev/crypto/tls#Config) so if you load in the RootCA that you created the self-signed certificates from, then they'll be trusted and you shouldn't need to disable verification if you have the SubjectAlternativeNames populated with the hostname(s) / ip addresses you want to connect to your backend using
